web_ui/frontend/components/CardList.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {ComponentType, FunctionComponent, useMemo, useState, JSX} from "react";
+import {Box, Pagination, Skeleton, TextField, Typography} from "@mui/material";
+
+interface CardListProps<T> {
+    data?: Partial<T>[];
+    Card: ComponentType<any>;
+    cardProps: Partial<T>;
+}
+
+export function CardList<T>({ data, Card, cardProps }: CardListProps<T>) {
+
+    const PAGE_SIZE = 5
+    const [page, setPage] = useState<number>(1)
+
+    const count = useMemo(() => {
+        return Math.ceil((data?.length || 0) / PAGE_SIZE)
+    }, [data])
+
+    const slicedObjects = useMemo(() => {
+        return (data || []).slice((page - 1) * PAGE_SIZE, page * PAGE_SIZE)
+    }, [data, page])
+
+    return (
+        <>
+            <Box>
+                {slicedObjects.map((o, i) => {
+
+                    const props = {
+                        ...cardProps,
+                        ...o
+                    } as T
+
+                    return <Box pb={1} key={JSON.stringify(props)}>
+                        <Card {...props} />
+                    </Box>
+                })}
+                {!data || count == 0 && <Skeleton variant="rectangular" height={60}></Skeleton>}
+            </Box>
+            <Box display={"flex"} justifyContent={"center"} flexDirection={"column"} alignItems={"center"}>
+                { count > 1 && <Pagination count={count} page={page} onChange={(e, p) => setPage(p)} /> }
```

Comment:
This count seems to be off. Without filtering anything, I got 5 entries in one page, but the number shows 8.

![Screenshot 2024-06-11 at 10 05 26 AM](https://github.com/PelicanPlatform/pelican/assets/41393704/beb8dae7-4d04-4442-b05a-d9af88cf4ecc)
I thought the real "count" here is the total number of servers in the table, as we don't really need a count for servers per page.

Looking at MUI docs, `count` is referring to "total number of pages": https://mui.com/material-ui/api/pagination/